rest/oidc_api_test.go

```diff
@@ -0,0 +1,80 @@
+package rest
+
+import (
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/couchbase/sync_gateway/auth"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleOIDCCommon(t *testing.T) {
```

Comment:
Testing a single handler doesn't seem appropriate for this feature, it seems like you'd need a bit more of a functional test to:

1. Send the initial OIDC request and have state cookie set on the client
2. Have a provider return the state to you via the callback URL
3. Have the code you wrote in handleOIDCCallback ensure they match (or don't)
4. Assert based on status code/error returned by the callback request

Include a way to change or remove the state cookie between steps 1 and 3 to make sure the error cases are covered.